apps/user-portal/src/components/shared/modal.tsx

```diff
@@ -0,0 +1,131 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { MouseEvent } from "react";
+import { Button, Icon, Modal } from "semantic-ui-react";
+
+interface ModalProps {
+    type: "positive" | "negative" | "warning" | "info";
+    heading?: string;
+    description: string;
+    children?: React.ReactNode;
+    open: any;
+    onClose: any;
+    primaryAction?: string;
+    secondaryAction?: string;
+    onPrimaryActionClick?: () => void;
+    onSecondaryActionClick?: (e: MouseEvent<HTMLElement>) => void;
+}
+
+const ModalComponentInner = (props: ModalProps) => {
+    const {
+        children,
+        type,
+        heading,
+        description,
+        open,
+        onClose,
+        primaryAction,
+        secondaryAction,
+        onPrimaryActionClick,
+        onSecondaryActionClick
+    } = props;
+
+    const iconName = () => {
+        if (type === "positive") {
+            return (
+            <div className="sa">
+                <div className="sa-success">
+                    <div className="sa-success-tip"/>
+                    <div className="sa-success-long"/>
+                    <div className="sa-success-placeholder"/>
+                    <div className="sa-success-fix"/>
+                </div>
+            </div>
+            );
+        } else if (type === "negative") {
+            return (
+                <div className="sa">
+                <div className="sa-error">
+                    <div className="sa-error-x">
+                        <div className="sa-error-left"/>
+                        <div className="sa-error-right"/>
+                    </div>
+                    <div className="sa-error-placeholder"/>
+                    <div className="sa-error-fix"/>
+                </div>
+            </div>
+            );
+        } else if (type === "warning") {
+            return (
+                    <Icon
+                        className="modal-icon"
+                        name="exclamation circle"
+                        size="huge"
+                        color="yellow"
+                    />
+            );
+        } else {
+            return (
+                    <Icon
+                        className="modal-icon"
+                        name="info circle"
+                        size="huge"
+                        color="blue"
+                    />
+            );
+        }
+    };
+
+    return (
+        <Modal
+            size="mini"
+            className="custom-modal"
+            dimmer="blurring"
+            open={ open }
+            onClose={ onClose }
```

Comment:
we better remove size and dimmer from here and spread the unhandled props. If the modal size & dimmer has to be `mini` & `blurring` by default, let's add them in defaultProps.